Modern/wrapperPlugins/src/main/java/org/bonej/wrapperPlugins/wrapperUtils/UsageReporterOptions.java

```diff
@@ -0,0 +1,135 @@
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+package org.bonej.wrapperPlugins.wrapperUtils;
+
+import java.util.Random;
+
+import net.imagej.ImageJ;
+
+import org.scijava.ItemVisibility;
+import org.scijava.command.Command;
+import org.scijava.command.ContextCommand;
+import org.scijava.plugin.Parameter;
+import org.scijava.plugin.Plugin;
+import org.scijava.widget.Button;
+
+/**
+ * Handles persistent settings such as user permission state.
+ * 
+ * Preferences are stored in their native format (long, int, boolean, etc.)
+ * 
+ * @author Michael Doube
+ * @author Richard Domander
+ *
+ */
+@Plugin(type = Command.class, menuPath = "Edit>Options>BoneJ Usage (Modern)")
+public class UsageReporterOptions extends ContextCommand {
+
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message1 = "Allow usage data collection?";
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message2 = "BoneJ would like to collect data on";
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message3 = "which plugins are being used, to direct development"; 
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message4 = "and promote BoneJ to funders.";
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String message5 = "If you agree to participate please check the box.";		
+	@Parameter(visibility = ItemVisibility.MESSAGE)
+	private String helpMessage = "For more information click Help.";
+	
+	@Parameter(label = "Opt in to usage data collection", description = "Can BoneJ send usage data?")
+  private boolean optIn = false;
+	
+	@Parameter(label = "Help")
+	private Button button;
+	
+	@Parameter
+	private ImageJ imagej;
+
+	/** set to true if user permission has been requested */
+	static final String OPTINSET = "bonej2.report.option.set";
+	/** Set to false if reporting is not allowed */
+	static final String OPTINKEY = "bonej2.allow.reporter";
+	static final String COOKIE = "bonej2.report.cookie";
+  static final String COOKIE2 = "bonej2.report.cookie2";
+  /** time of first visit in seconds */
+	static final String FIRSTTIMEKEY = "bonej2.report.firstvisit";
+	/** unique ID for this particular BoneJ session */
+	static final String SESSIONKEY = "bonej2.report.bonejsession";
+	static final String IJSESSIONKEY = "bonej2.report.ijsession";
+
+	@Override
+	public void run() {
+
+		//TODO link to detailed online info
+    //For more information click Help.";
+		
+		//Wipe persistent data on opt-out
+		if (!optIn) {
+			System.out.print("User has opted out of data collection\n");
+			imagej.prefs().clear(this.getClass());
```

Comment:
`return` missing :open_mouth:

Reply:
OK good point. Now it returns from the `if (!optIn)` block. Fixed in 9ffacd5d

Bear in mind that we need to keep a setting `OPTINSET` that shows user permission has been sought, otherwise users who opted out will get bugged over and over whether they want to opt in or out.